pkg/storage/etcd3/watcher.go

```diff
@@ -322,7 +322,13 @@ func prepareObjs(ctx context.Context, e *event, client *clientv3.Client, codec r
 		if err != nil {
 			return nil, nil, err
 		}
-		oldObj, err = decodeObj(codec, versioner, getResp.Kvs[0].Value, getResp.Kvs[0].ModRevision)
+		rev := getResp.Kvs[0].ModRevision
+		if e.isDeleted {
+			// Note that this sends the *old* object with the etcd revision for the time at
+			// which it gets deleted. Users use that revision to tell ordering.
+			rev = e.rev
+		}
+		oldObj, err = decodeObj(codec, versioner, getResp.Kvs[0].Value, rev)
```

Comment:
this is strange... why shall we change the revision of the old object? for the watch issue, can you show me the code for etcd2? i would assume there are two objs in the event curObj and oldObj and e.Rev should be the rev of curObj. So the rev of the oldObj should not matter at all for ordering purpose. No?